src/llvm/lib_llvm.cr

```diff
@@ -1,26 +1,6 @@
 {% begin %}
 lib LibLLVM
-  LLVM_CONFIG = {{
-                  `[ -n "$LLVM_CONFIG" ] && command -v "$LLVM_CONFIG" || \
-                   command -v llvm-config-8 || command -v llvm-config-8.0 || command -v llvm-config80 || \
-                   (command -v llvm-config > /dev/null && (case "$(llvm-config --version)" in 8.0*) command -v llvm-config;; *) false;; esac)) || \
-                   command -v llvm-config-7 || \
-                   (command -v llvm-config > /dev/null && (case "$(llvm-config --version)" in 7.1*) command -v llvm-config;; *) false;; esac)) || \
-                   command -v llvm-config-7.0 || command -v llvm-config70 || \
-                   (command -v llvm-config > /dev/null && (case "$(llvm-config --version)" in 7.0*) command -v llvm-config;; *) false;; esac)) || \
-                   command -v llvm-config-6.0 || command -v llvm-config60 || \
-                   (command -v llvm-config > /dev/null && (case "$(llvm-config --version)" in 6.0*) command -v llvm-config;; *) false;; esac)) || \
-                   command -v llvm-config-5.0 || command -v llvm-config50 || \
-                   (command -v llvm-config > /dev/null && (case "$(llvm-config --version)" in 5.0*) command -v llvm-config;; *) false;; esac)) || \
-                   command -v llvm-config-4.0 || command -v llvm-config40 || \
-                   (command -v llvm-config > /dev/null && (case "$(llvm-config --version)" in 4.0*) command -v llvm-config;; *) false;; esac)) || \
-                   command -v llvm-config-3.9 || command -v llvm-config39 || \
-                   (command -v llvm-config > /dev/null && (case "$(llvm-config --version)" in 3.9*) command -v llvm-config;; *) false;; esac)) || \
-                   command -v llvm-config-3.8 || command -v llvm-config38 || \
-                   (command -v llvm-config > /dev/null && (case "$(llvm-config --version)" in 3.8*) command -v llvm-config;; *) false;; esac)) || \
-                   command -v llvm-config
-                  `.chomp.stringify
-                }}
+  LLVM_CONFIG = {{ system(read_file("#{__DIR__}/../../bin/find-llvm-config")).stringify }}
```

Comment:
Unfortunately, `bin/find-llvm-config` wouldn't be included in most distribution packages of crystal. Not even in our own ones. This would need an update in https://github.com/crystal-lang/distribution-scripts

So currently this code only works in the compiler repository. Using the libllvm bindings outside the compiler would fail. Since `LLVM` is public API, it needs to work anywhere.

One way to solve this is updating distribution-scripts and making sure all other package distributors make this change, too. Maybe there should be a simple rule that all files in `bin` should be included in the package?
The other option would be to move this helper script to `src/llvm/find-llvm-config`. This is obviously easier to achieve, and probably even a better solution, because it clearly belongs to the LLVM module.

Reply:
Good catch, this will be one less thing to do for the packagers.

I will move the script to a `src/llvm/helpers` directory, in order to not have shell scripts along with Crystal code, a bit like the existing `ext`.

Reply:
I'd be happy to put it in `src/llvm/ext`.